doc/tutorials/content/executor.rst

```diff
@@ -0,0 +1,102 @@
+.. _executor:
+
+Using Executors
+--------------------------------------------------
+
+Executors allows control over the where and how a piece of program executes
```

Comment:
Executors help in providing a fine-grained control to the programmer to choose where and how a piece of program will execute. All functions supporting the executors have a uniform API for running the program in multiple ways

(or something like this)